protocol/messenger.go

```diff
@@ -146,6 +146,7 @@ type Messenger struct {
 	cancel context.CancelFunc
 
 	importingCommunities map[string]bool
+	importingChannels    map[string]bool
```

Comment:
I'm wondering whether this map is necessary...
The equivalent `importingCommunities` was originally introduce to keep track of whether an import of a community was marked as cancelled. And because there could be theoretically multiple imports happen at the same time, we needed a map for that.

The UI however, restricts this to only having one community import at a time (which actually means there's no need for a map in the first place, but it was already there and I think there were discussions around this actually being enabled for multiple imports running in parallel).

The `importingChannels` map hence implies that there can be multiple channels being imported into various communities at the same time.

Do we allow for that in the UI?

Also, I see we use `DiscordImportChannelMarkedAsCancelled` in various places where we used to us `DiscordImportMarkedAsCancelled`.

Notice that we used to initialize a map entry in `importingCommunities` at the beginning of each import.
This was to ensure so we can mark it as cancelled and cleanup work afterwards.

Cleanup particularly happens in https://github.com/status-im/status-go/pull/4160/files#diff-e8ae84ba629f58c08abd2b2b3864896c92dd9b3898cf7b450d378f76004bf312R5371-R5391

I'm seeing most of the `markDiscordCommunityImportAsCancelled()` calls have been removed (and replaced with the channel import equivalent). This is a problem because this potentially prevents the communityID entry in the `importingCommunity` map, which needs to exist to do the `cleanup()`.

Otherwise you keep creating zombie communities.

So I think we need to figure out whether the import has originally started as "community import" and if yes, then we need to mark it as such.

Reply:
> The importingChannels map hence implies that there can be multiple channels being imported into various communities at the same time.
> Do we allow for that in the UI?

No we don't. At one point we will like to support that, but we block it to keep it simple for now.

> Also, I see we use DiscordImportChannelMarkedAsCancelled in various places where we used to us DiscordImportMarkedAsCancelled.

I think `DiscordImportMarkedAsCancelled` is still there. Boris just duplicated a lot of the code to make importing a channel work. I don't see code deletion, so the old code for the community import should still be all there.

I added an issue to clean up this code because of that https://github.com/status-im/status-desktop/issues/12503

I could be wrong though. Boris is off until Thursday, that's why I'm answering, since I want to move this along and merged soon.

Reply:
> I think DiscordImportMarkedAsCancelled is still there. Boris just duplicated a lot of the code to make importing a channel work. I don't see code deletion, so the old code for the community import should still be all there.

That is a very good point. I was searching through the changes and could only see one usage of `MarkDiscordImportCommunityAsCancelled` (I think in handling authors data code, that's why I thought it was removed everywhere else.

But you're absolutely right, this is only adding stuff.

Ignore my comment then